src/components/legend/style.js

```diff
@@ -241,8 +242,49 @@ module.exports = function style(s, gd) {
         txt.selectAll('text').call(Drawing.textPointStyle, tMod, gd);
     }
 
+    function styleWaterfalls(d) {
```

Comment:
![image](https://user-images.githubusercontent.com/6675409/55027044-b20a6000-4fda-11e9-9c9b-2085ebe6e979.png)

Could we not show the "totals" part in the legend symbol when the trace only shows increasing/decreasing bars.